utils/Api.php

```diff
@@ -173,14 +173,14 @@ public function getPageAsync( $title ) {
 
 	private function parseGetPageResponse( $response ) {
 		foreach ( $response['query']['pages'] as $page ) {
+			$title = $page['title'];
 			if ( isset( $page['revisions'] ) ) {
 				foreach ( $page['revisions'] as $revision ) {
-					return getXhtmlFromContent( $this->lang, $revision['*'], $page['title'] );
+					return getXhtmlFromContent( $this->lang, $revision['*'], $title );
 				}
 			}
 		}
-
-		throw new HttpException( 'Page revision not found', 404 );
+		throw new HttpException( "Page revisions not found for: $title", 404 );
```

Comment:
The integration test fails because `BookTest::testGetNonExistingTitleDisplaysError` expects the error message to be "Page revision not found" with revision singular.

Reply:
Fixed.